### ISSUE TITLE
attaching multiple loss

### ISSUE THREAD

**Issue by colmantse** [tensorflow/tensor2tensor#952]
Dear all,

I would like to confirm if i want to have multiple losses to minimize, do I simply attached the loss in the loss dict? so the loss would be accumulated and taken account of via summation in the t2t_model estimator_model_fn? 

as in:

    # Accumulate losses
    loss = sum(losses_dict[key] for key in sorted(losses_dict.keys()))

I wonder if this is the standard procedure. It seems that all all loss together to form a single total loss. I traced it to the optimize.optimize, where the loss there seems to be the "total loss"

On the other hand, I read about writing multiple loss with estimator here . 
So multi-loss can also be implemented in this manner, tf.group (train_ops_1, train_ops_2), 

If so, do I start with overwriting estimator_spec_train primarily and probably overwrite optimize.optimize to lossen the variable name "training_loss", "total_loss" constraints ?

Would be great if anyone can give me some directions.

Warm regards,
Colman

**Comment by cbailes** [comment]
[attaching multiple loss](https://groups.google.com/d/msg/tensor2tensor/W0422BVl_H0/eBW3nytpDgAJ) @ tensor2tensor Google Group
